src/System.Windows.Extensions/src/System/Drawing/ImageConverter.cs

```diff
@@ -108,33 +108,32 @@ private unsafe Stream GetBitmapStream(ReadOnlySpan<byte> rawData)
         {
             try
             {
+                short signature = MemoryMarshal.Read<short>(rawData);
+
+                if (signature != 0x1c15 || rawData.Length < sizeof(OBJECTHEADER))
+                {
+                    return null;
+                }
+
                 // The data is in the form of OBJECTHEADER. It's an encoded format that Access uses to push imagesinto the DB.
                 OBJECTHEADER pHeader = MemoryMarshal.Read<OBJECTHEADER>(rawData);
 
                 // pHeader.signature will always be 0x1c15.
                 // "PBrush" should be the 6 chars after position 12 as well.
-                if (pHeader.signature != 0x1c15 ||
-                    rawData.Length <= pHeader.headersize + 18 ||
+                if ( rawData.Length <= pHeader.headersize + 18 ||
                     !rawData.Slice(pHeader.headersize + 12, 6).SequenceEqual(PBrush))
                 {
                     return null;
                 }
 
                 // We can safely trust that we've got a bitmap.
-                // search for "BM" in the data which is the start of our bitmap data.
-                // 18 is from (12+6) above.
-                ReadOnlySpan<byte> rawDataWithoutHeader = rawData.Slice(pHeader.headersize + 18);
-                int index = rawDataWithoutHeader.IndexOf(BMBytes);
-
-                if (index >= 0 && index < pHeader.headersize + 510)
-                {
-                    return new MemoryStream(rawDataWithoutHeader.Slice(index).ToArray());
-                }
+                // The start of our bitmap data in the rawdata is always 78.
+                return new MemoryStream(rawData.Slice(78).ToArray());
```

Comment:
@stephentoub, should there ideally be a MemoryStream constructor that accepts a span?

Reply:
>  should there ideally be a MemoryStream constructor that accepts a span?

No; that would suggest that the MemoryStream could store the span without making a copy, but it can't.